pkg/query-service/postprocess/table.go

```diff
@@ -0,0 +1,299 @@
+			orderByList := query.OrderBy
+			if len(orderByList) == 0 {
+				// If no orderBy is specified, sort by value in descending order
+				orderByList = []v3.OrderBy{{ColumnName: constants.SigNozOrderByValue, Order: "desc"}}
+			}
+			for _, orderBy := range orderByList {
+				name := orderBy.ColumnName
+				if name == constants.SigNozOrderByValue {
+					name = queryName
+				}
+				colIndex := -1
+				for k, col := range columns {
+					if col.Name == name {
+						colIndex = k
+						break
+					}
+				}
+				if colIndex == -1 {
+					continue
+				}
+
+				valI := rows[i].Data[colIndex]
+				valJ := rows[j].Data[colIndex]
+
+				// Handle "n/a" values
+				if valI == "n/a" && valJ == "n/a" {
+					continue
+				}
+
+				// Compare based on the data type
+				switch v := valI.(type) {
+				case float64:
+					switch w := valJ.(type) {
+					case float64:
+						if v != w {
+							return (v < w) == (orderBy.Order == "asc")
+						}
+					default:
+						// For any other type, sort float64 first
+						return orderBy.Order == "asc"
+					}
+				case string:
+					switch w := valJ.(type) {
+					case float64:
+						// If types are different, sort numbers before strings
+						return orderBy.Order != "asc"
+					case string:
+						if v != w {
+							return (v < w) == (orderBy.Order == "asc")
+						}
+					default:
+						// For any other type, sort strings before bools
+						return orderBy.Order == "asc"
+					}
+				case bool:
+					switch w := valJ.(type) {
+					case float64, string:
+						// If types are different, sort bools after numbers and strings
+						return orderBy.Order != "asc"
+					case bool:
+						if v != w {
+							return (!v && w) == (orderBy.Order == "asc")
+						}
+					}
+				}
+			}
+		}
+		return false
+	})
+}
+
+func TransformToTableForClickHouseQueries(results []*v3.Result) []*v3.Result {
+	if len(results) == 0 {
+		return []*v3.Result{}
+	}
+
+	// Sort results by QueryName
+	sort.Slice(results, func(i, j int) bool {
+		return results[i].QueryName < results[j].QueryName
+	})
+
+	// Create a map to store all unique labels
+	seen := make(map[string]struct{})
+	labelKeys := []string{}
+	for _, result := range results {
+		for _, series := range result.Series {
+			for _, labels := range series.LabelsArray {
+				for key := range labels {
+					if _, ok := seen[key]; !ok {
+						seen[key] = struct{}{}
+						labelKeys = append(labelKeys, key)
+					}
+				}
+			}
+		}
+	}
+
+	// Create columns
+	// Why don't we have a column for each query name?
+	// Because we don't know if the query is an aggregation query or a non-aggregation query
+	// So we create a column for each query name that has at least one point
+	columns := make([]*v3.TableColumn, 0)
+	for _, key := range labelKeys {
+		columns = append(columns, &v3.TableColumn{Name: key})
+	}
+	for _, result := range results {
+		if len(result.Series) > 0 && len(result.Series[0].Points) > 0 {
+			columns = append(columns, &v3.TableColumn{Name: result.QueryName})
```

Comment:
if there are multiple result columns for the same query then we would be adding duplicates here ?

Reply:
It's not possible to have multiple result columns in the same query as of now. There can only be one result column. Although this will be changed soon but it will be separately addressed.